app/common/utils.js

```diff
@@ -0,0 +1,7 @@
+export function shiftItemRandomly(array) {
+    return array.splice(getRandomInt(0, array.length), 1)[0];
```

Comment:
This method is hard to understand and to validate. Try to make it simpler. It has side effects (splice has, while slice has not). If that´s your intention, then shift is not the best name. I´d expect shiftItemRandomly([1, 2, 3]) to generate [1, 3, 2] (3 was shift left).